### PR TITLE
meta: migrate SSCCE to TypeScript

### PR DESCRIPTION
**Pull request by ephys**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

The `sscce.js` file in this repository has been broken for a while as it tries to import TypeScript files.

This fixes that by migrating it to TypeScript and using ts-node to run it

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "./packages/core/test/tsconfig.json",
+  "compilerOptions": {
+    "rootDir": "..",
+    "module": "Node16"
```
Comment (WikiRik) [inline tsconfig.json:5]: Any specific reason to go for Node 16?

File: dev/sscce-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import type { Options, Sequelize } from '../packages/core';
+import * as Support from '../packages/core/test/support';
```
Comment (WikiRik) [inline dev/sscce-helpers.ts:2]: Why not import `createSequelizeInstance` directly? Does that give naming conflict issues?

**Review by WikiRik — APPROVED** [review]
Two small remarks but should be good. Thanks for working on this!

File: tsconfig.json (hunk 1 above)
Comment (ephys) [reply tsconfig.json:5]: That's from I tried to make sscce.ts an ESM module

File: dev/sscce-helpers.ts (hunk 2 above)
Comment (ephys) [reply dev/sscce-helpers.ts:2]: Yep! The method this file exports is also called `createSequelizeInstance`

File: tsconfig.json (hunk 1 above)
Comment (ephys) [reply tsconfig.json]: done

**Review by WikiRik — APPROVED** [review]

**Pull request merged**